### PR TITLE
Fix bug: lose monitored_dbid_cache after switching mirror

### PR DESCRIPTION
**Pull request by zhrt123**

### CODE REVIEW

File: src/monitored_db.c [hunk 1]
```diff
@@ -336,3 +336,29 @@ dump_monitored_dbid_cache(long *nitems)
...
+	}
+	if (!found)
+	{
+		entry->paused = false;
+		pg_atomic_init_u32(&(entry->epoch), 0);
+		pg_atomic_init_u32(&(entry->status), DB_RUNNING);
+	}
+
+	LWLockRelease(diskquota_locks.monitored_dbid_cache_lock);
+}
```
Comment (higuoxing) [inline src/monitored_db.c:364]: Nit: Put a `\newline` at the end of this file.

File: src/monitored_db.c [hunk 2]
```diff
@@ -336,3 +336,29 @@ dump_monitored_dbid_cache(long *nitems)
...
+	LWLockAcquire(diskquota_locks.monitored_dbid_cache_lock, LW_EXCLUSIVE);
+	MonitorDBEntry entry = hash_search(monitored_dbid_cache, &dbid, HASH_ENTER_NULL, &found);
+	if (entry == NULL)
+	{
+		ereport(WARNING, (errmsg("can't alloc memory on dbid cache, there ary too many databases to monitor")));
+	}
+	if (!found)
+	{
+		entry->paused = false;
+		pg_atomic_init_u32(&(entry->epoch), 0);
```
Comment (higuoxing) [inline src/monitored_db.c:359]: The `monitored_dbid_cache` has been guarded by `LWLock`, do we still need `pg_atomic_init_u32` to initialize it?

**Review by higuoxing — APPROVED** [review]
Looks good with some minor comments.

File: src/monitored_db.c [hunk 3]
```diff
@@ -336,3 +336,29 @@ dump_monitored_dbid_cache(long *nitems)
...
+void
+refresh_monitored_dbid_cache(void)
+{
+	bool found;
+	Oid  dbid = MyDatabaseId;
+	LWLockAcquire(diskquota_locks.monitored_dbid_cache_lock, LW_EXCLUSIVE);
+	MonitorDBEntry entry = hash_search(monitored_dbid_cache, &dbid, HASH_ENTER_NULL, &found);
+	if (entry == NULL)
+	{
+		ereport(WARNING, (errmsg("can't alloc memory on dbid cache, there ary too many databases to monitor")));
```
Comment (yihong0618) [inline src/monitored_db.c:354]: there ary -> there are

Comment (yihong0618) [inline src/monitored_db.c:354]: should we early return and release the lock here?

File: src/monitored_db.c [hunk 4]
```diff
@@ -336,3 +336,29 @@ dump_monitored_dbid_cache(long *nitems)
...
+	}
+	if (!found)
+	{
+		entry->paused = false;
+		pg_atomic_init_u32(&(entry->epoch), 0);
+		pg_atomic_init_u32(&(entry->status), DB_RUNNING);
```
Comment (beeender) [inline src/monitored_db.c:360]: Add a log here then we are good to go!

**Review by beeender — APPROVED** [review]

File: src/monitored_db.c (hunk 2 above)
Comment (zhrt123) [reply src/monitored_db.c:359]: We just can initialize it by `pg_atomic_init_u32`, we cannot set it to 0 by `=`.

File: src/monitored_db.c (hunk 3 above)
Comment (zhrt123) [reply src/monitored_db.c:354]: thanks!

**Pull request merged**